### ISSUE TITLE
sh: Can't open winetricks

### ISSUE THREAD

**Issue by Golddouble** [alessivs/supermemo-wine#7]
# Platform information
MX Linux 19 (Debian, Xfce)

## Name of checkout or release of supermemo-wine used
supermemo17.verb
Wine 5.2
Winetricks 0.0+20181203-3

When I open my terminal and type in :
`env WINEARCH=win32 sh winetricks arch=32 prefix=supermemo17`

I get the following Output:
`sh: 0: Can't open winetricks`

Seems not to work. (?)

**Comment by alessivs** [comment]
Try this:

```
env WINEARCH=win32 sh /usr/bin/winetricks arch=32 prefix=supermemo17
```

**Comment by alessivs** [comment]
Note that it's an old winetricks script for a rather new Wine release. If you run into a runtime issue where it is not possible to view any HTML components in SuperMemo, download the most recent winetricks release, which includes a workaround for one of the required IE8 dlls in recent Wine. Example:

```sh
# download the latest release to the current directory
wget https://raw.githubusercontent.com/Winetricks/winetricks/20191224/src/winetricks 
# run winetricks from the current directory
env WINEARCH=win32 sh ./winetricks arch=32 prefix=supermemo17
sh ./winetricks prefix=supermemo17 supermemo17.verb
```

In the future there will be a wrapper that automatically downloads the latest release of Winetricks.

**Comment by Golddouble** [comment]
Thank you.
I have installed now winetricks from the Test Repository of my Linux distrubution. So this way I have version 20191224-0.

In the meantime I found out, that winetricks is not in 
/usr/bin/winetricks but it is in the root:
/bin/winetricks

(Just have installed it with my Linux distribution packet manager and not manipulated anything.)
I have no .bin Folder in my home.

**Question 1**: Is this a problem? When Winetricks is installed in root?

So I tried this:
```
VM-Konto@mx:/bin
$ env WINEARCH=win32 sh /bin/winetricks arch=32 prefix=supermemo17
------------------------------------------------------
WINEPREFIX INFO:
Drive C: insgesamt 32
drwxr-xr-x  7 VM-Konto VM-Konto 4096 Apr  5 16:34 .
drwxr-xr-x  4 VM-Konto VM-Konto 4096 Apr  6 16:24 ..
drwxr-xr-x  3 VM-Konto VM-Konto 4096 Apr  5 16:34 ProgramData
drwxr-xr-x  6 VM-Konto VM-Konto 4096 Apr  5 16:33 Program Files
drwxr-xr-x  6 VM-Konto VM-Konto 4096 Apr  5 16:34 Program Files (x86)
drwxr-xr-x  4 VM-Konto VM-Konto 4096 Apr  5 16:34 users
drwxr-xr-x 18 VM-Konto VM-Konto 4096 Apr  5 16:34 windows
-rw-r--r--  1 VM-Konto VM-Konto    9 Apr  5 16:33 .windows-serial

Registry info:
/home/VM-Konto/.wine/system.reg:#arch=win64
/home/VM-Konto/.wine/user.reg:#arch=win64
/home/VM-Konto/.wine/userdef.reg:#arch=win64
------------------------------------------------------
------------------------------------------------------
wine cmd.exe /c echo '%ProgramFiles%' returned empty string, error message "wine: WINEARCH set to win32 but '/home/VM-Konto/.wine' is a 64-bit installation." 
------------------------------------------------------
VM-Konto@mx:/bin
```

Regarding to "wine":
This is what I have installed from my Repository. I do not see a 32 bit version here:
https://www.bilder-upload.eu/bild-ebcdf3-1586246930.png.html
(This picture is before I changed to Winetricks 20191224-0)

I thought 64bit wine can also deal with 32 bit applications. Not true?
**Question 2:** So it seems I need to avoid somehow the error message above
`...returned empty string, error message...`

Would appreciate some help. Thank you.

**Comment by alessivs** [comment]
Ah distros. Why can't they agree. This could be a catch-all replacement:

```sh
env WINEARCH=win32 sh $(which winetricks) arch=32 prefix=supermemo17
```

The problem above, however, is apparently that winetricks isn't setting the `WINEPREFIX` environment variable on its own, which is why it's falling back to the default wine prefix located at `$HOME/.wine`. It has been suggested the problem has been fixed at the tip of the Winetricks master tree (see [this commit](https://github.com/Winetricks/winetricks/commit/e58973e215eef260a5b42acebba57fe5d109a2c9)). I don't have infinite time to monitor what breaks when. In the future I'll set up the environment variables myself in an installer wrapper. I'll suggest a command to run at the end of this message.

Regarding your wine question, you should be fine with the `wine` package installed. Packagers tend to omit the "32" from the package name, and only include it for the 64bit binary. (They tend to be installed together so I wouldn't be surprised if one is a dependency of the other.)

Regarding the 64bit arch question, winetricks will bail out from installing IE8 on such architecture[[1]](https://github.com/Winetricks/winetricks/blob/20191224/src/winetricks#L14371). Some packages just don't support 64bit, and this IE8 is apparently the case[[2]](https://github.com/Winetricks/winetricks/blob/20191224/src/winetricks#L460).

So think of a clean folder for your SuperMemo wine prefix. Don't create it, but make sure the parent folder exists (e.g. if you choose `~/.local/share/wineprefixes/sm17` ensure `~/.local/share/wineprefixes/` exists). Run the following command (replace paths as needed):

```sh
env WINEARCH=win32 WINEPREFIX=~/.local/share/wineprefixes/sm17 sh $(which winetricks) supermemo17.verb
```

**Comment by Golddouble** [comment]
Thank you. 
With your help I came until the step, where I had to download sm17inst.exe from https://super-memo.com and put it into  /home/VM-Konto/.cache/winetricks/supermemo17 and the browser popped out.

It seems that I was to slow. So the process aborted. 
In the meantime I have sm17inst.exe put into the folder.

It is said in the log of the terminal:

> Please download sm17inst.exe from https://super-memo.com and put it into  /home/VM-Konto/.cache/winetricks/supermemo17. Then repeate this command.

Question: Which command is it asking for?
I tried: 
`Running /usr/bin/wineserver -w`
But this was not successful.

Please note. There are some other errors in the terminal log. I do not know, if they are relevant(?)

The log from the terminal:
```
VM-Konto@mx:~
$ env WINEARCH=win32 WINEPREFIX=~/.local/share/wineprefixes/sm17 sh $(which winetricks) ~/Videos/supermemo17.verb
Using winetricks 20191224 - sha256sum: 59426092c150969fb9b4eb1444e8b0b931fa5525f6679cf94fcd4954fb1c03bd with wine-5.2 (Staging) and WINEARCH=win32
Executing w_do_call supermemo17
Executing load_supermemo17 
Executing w_do_call ie8
Executing load_ie8 
Setting Windows version to winxp
Executing wine regedit C:\windows\Temp\set-winver.reg
------------------------------------------------------
Running /usr/bin/wineserver -w. This will hang until all wine processes in prefix=/home/VM-Konto/.local/share/wineprefixes/sm17 terminate
------------------------------------------------------
Using builtin override for following DLLs: iexplore.exe
Executing wine regedit C:\windows\Temp\override-dll.reg
000b:fixme:winediag:start_process Wine Staging 5.2 is a testing version containing experimental patches.
000b:fixme:winediag:start_process Please mention your exact version when filing bug reports on winehq.org.
Executing wine iexplore -unregserver
Executing w_do_call msls31
Executing load_msls31 
Executing mkdir -p /home/VM-Konto/.cache/winetricks/msls31
Executing cd /home/VM-Konto/.cache/winetricks/msls31
Downloading https://ftp.hp.com/pub/softlib/software/msi/InstMsiW.exe to /home/VM-Konto/.cache/winetricks/msls31
--2020-04-07 14:07:36--  https://ftp.hp.com/pub/softlib/software/msi/InstMsiW.exe
Auflösen des Hostnamens ftp.hp.com (ftp.hp.com)… 15.73.48.59, 15.72.188.130
Verbindungsaufbau zu ftp.hp.com (ftp.hp.com)|15.73.48.59|:443 … verbunden.
HTTP-Anforderung gesendet, auf Antwort wird gewartet … 301 Moved Permanently
Platz: https://whp-aus1.cold.extweb.hp.com/pub/softlib/software/msi/InstMsiW.exe [folgend]
--2020-04-07 14:07:37--  https://whp-aus1.cold.extweb.hp.com/pub/softlib/software/msi/InstMsiW.exe
Auflösen des Hostnamens whp-aus1.cold.extweb.hp.com (whp-aus1.cold.extweb.hp.com)… 15.73.97.76, 15.73.97.75
Verbindungsaufbau zu whp-aus1.cold.extweb.hp.com (whp-aus1.cold.extweb.hp.com)|15.73.97.76|:443 … verbunden.
HTTP-Anforderung gesendet, auf Antwort wird gewartet … 200 OK
Länge: 1822848 (1.7M) [application/octet-stream]
Wird in »InstMsiW.exe« gespeichert.

InstMsiW.exe                  100%[===============================================>]   1.74M  1.03MB/s    in 1.7s    

2020-04-07 14:07:39 (1.03 MB/s) - »InstMsiW.exe« gespeichert [1822848/1822848]

Executing cd /home/VM-Konto
Executing cabextract -q --directory=/home/VM-Konto/.local/share/wineprefixes/sm17/dosdevices/c:/windows/temp /home/VM-Konto/.cache/winetricks/msls31/InstMsiW.exe
Executing cp -f /home/VM-Konto/.local/share/wineprefixes/sm17/dosdevices/c:/windows/temp/msls31.dll /home/VM-Konto/.local/share/wineprefixes/sm17/dosdevices/c:/windows/system32
Using native,builtin override for following DLLs: ieframe ieproxy itircl itss jscript msctf mshtml shdoclc shdocvw shlwapi urlmon wininet xmllite
Executing wine regedit C:\windows\Temp\override-dll.reg
000b:fixme:winediag:start_process Wine Staging 5.2 is a testing version containing experimental patches.
000b:fixme:winediag:start_process Please mention your exact version when filing bug reports on winehq.org.
Using native override for following DLLs: iexplore.exe
Executing wine regedit C:\windows\Temp\override-dll.reg
Using builtin override for following DLLs: updspapi
Executing wine regedit C:\windows\Temp\override-dll.reg
Executing mkdir -p /home/VM-Konto/.cache/winetricks/ie8
Executing cd /home/VM-Konto/.cache/winetricks/ie8
Downloading https://github.com/Winetricks/winetricks/raw/master/files/winetest.cat to /home/VM-Konto/.cache/winetricks/ie8
--2020-04-07 14:07:41--  https://github.com/Winetricks/winetricks/raw/master/files/winetest.cat
Auflösen des Hostnamens github.com (github.com)… 140.82.118.4
Verbindungsaufbau zu github.com (github.com)|140.82.118.4|:443 … verbunden.
HTTP-Anforderung gesendet, auf Antwort wird gewartet … 302 Found
Platz: https://raw.githubusercontent.com/Winetricks/winetricks/master/files/winetest.cat [folgend]
--2020-04-07 14:07:41--  https://raw.githubusercontent.com/Winetricks/winetricks/master/files/winetest.cat
Auflösen des Hostnamens raw.githubusercontent.com (raw.githubusercontent.com)… 151.101.112.133
Verbindungsaufbau zu raw.githubusercontent.com (raw.githubusercontent.com)|151.101.112.133|:443 … verbunden.
HTTP-Anforderung gesendet, auf Antwort wird gewartet … 200 OK
Länge: 448 [application/octet-stream]
Wird in »winetest.cat« gespeichert.

winetest.cat                  100%[===============================================>]     448  --.-KB/s    in 0s      

2020-04-07 14:07:41 (4.24 MB/s) - »winetest.cat« gespeichert [448/448]

Executing cd /home/VM-Konto
Executing cp -f /home/VM-Konto/.cache/winetricks/ie8/winetest.cat /home/VM-Konto/.local/share/wineprefixes/sm17/dosdevices/c:/windows/system32/catroot/{f750e6c3-38ee-11d1-85e5-00c04fc295ee}/oem0.cat
Executing cd /home/VM-Konto/.cache/winetricks/ie8
Downloading https://download.microsoft.com/download/C/C/0/CC0BD555-33DD-411E-936B-73AC6F95AE11/IE8-WindowsXP-x86-ENU.exe to /home/VM-Konto/.cache/winetricks/ie8
--2020-04-07 14:07:41--  https://download.microsoft.com/download/C/C/0/CC0BD555-33DD-411E-936B-73AC6F95AE11/IE8-WindowsXP-x86-ENU.exe
Auflösen des Hostnamens download.microsoft.com (download.microsoft.com)… 23.54.115.197, 2a02:26f0:3000:298::e59, 2a02:26f0:3000:289::e59, ...
Verbindungsaufbau zu download.microsoft.com (download.microsoft.com)|23.54.115.197|:443 … verbunden.
HTTP-Anforderung gesendet, auf Antwort wird gewartet … 200 OK
Länge: 16883056 (16M) [application/octet-stream]
Wird in »IE8-WindowsXP-x86-ENU.exe« gespeichert.

IE8-WindowsXP-x86-ENU.exe     100%[===============================================>]  16.10M  11.8MB/s    in 1.4s    

2020-04-07 14:07:43 (11.8 MB/s) - »IE8-WindowsXP-x86-ENU.exe« gespeichert [16883056/16883056]

Executing cd /home/VM-Konto
Executing cd /home/VM-Konto/.cache/winetricks/ie8
0030:fixme:clusapi:GetNodeClusterState ((null),0031EC34) stub!
0030:fixme:advapi:DecryptFileA ("y:\\acd4ccdf7567d6b363863636ba\\", 00000000): stub
0038:fixme:ntdll:EtwRegisterTraceGuidsW (0x6cd15f38, 0x6cd20180, {e2821408-c59d-418f-ad3f-aa4e792aeb79}, 1, 0x31f5c0, (null), (null), 0x6cd20188): stub
0038:fixme:ntdll:EtwRegisterTraceGuidsW   register trace class {e2821408-c59d-418f-ad3f-aa4e792aeb79}
0038:fixme:wintrust:SOFTPUB_VerifyImageHash Cannot verify hash for pszObjId=(null)
0038:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
0038:fixme:richedit:ME_HandleMessage EM_SETMARGINS: stub
0038:fixme:wuapi:update_session_put_ClientApplicationID 00110D58, L"Windows Internet Explorer 8 Setup Utility"
003c:fixme:wuapi:automatic_updates_Pause 
003f:fixme:setupapi:pSetupGetGlobalFlags stub
003f:fixme:wintrust:WinVerifyTrust unimplemented for 4
003f:fixme:wintrust:SOFTPUB_VerifyImageHash Cannot verify hash for pszObjId=(null)
003f:err:ole:CoGetClassObject class {acadf079-cbcd-4032-83f2-fa47c4db096f} not registered
003f:err:ole:CoGetClassObject no class object {acadf079-cbcd-4032-83f2-fa47c4db096f} could be created for context 0x1
003f:fixme:ntdll:server_ioctl_file Unsupported ioctl 9c040 (device=9 access=3 func=10 method=0)
003f:fixme:sfc:SRSetRestorePointA 0031B914 0031B7B8
003f:fixme:sfc:SfcConnectToServer 0
003f:fixme:wintrust:CryptCATGetCatAttrInfo FFFFFFFF, L"SPAttr"
003f:fixme:wintrust:CryptCATGetCatAttrInfo 00CCB080, L"SPAttr"
003f:fixme:wintrust:CryptCATGetCatAttrInfo FFFFFFFF, L"SPLevel"
003f:fixme:wintrust:CryptCATGetCatAttrInfo 00CCB148, L"SPLevel"
003f:fixme:wintrust:SOFTPUB_VerifyImageHash Cannot verify hash for pszObjId=(null)
CreateService SUCCESS.
003f:fixme:setupapi:pSetupGetGlobalFlags stub
003f:fixme:setupapi:pSetupGetGlobalFlags stub
003f:fixme:wintrust:CryptCATGetCatAttrInfo 00529D00, L"SPAttr"
003f:fixme:wintrust:CryptCATGetCatAttrInfo 00529D00, L"SPAttr"
003f:fixme:wintrust:CryptCATGetCatAttrInfo 00529D00, L"SPLevel"
003f:fixme:wintrust:CryptCATGetCatAttrInfo 00529D00, L"SPLevel"
003f:fixme:wintrust:SOFTPUB_VerifyImageHash Cannot verify hash for pszObjId=(null)
003f:fixme:propsys:PSRegisterPropertySchema L"C:\\Program Files\\Internet Explorer\\ie8props.propdesc" stub
003f:fixme:advapi:TreeResetNamedSecurityInfoW (L"C:\\windows\\ie8", 1, 4, (nil), (nil), 0xcc9218, (nil), 1, (nil), 1, (nil)) stub
003f:fixme:advapi:RegisterEventSourceA ((null),"Internet Explorer 8"): stub
003f:fixme:advapi:RegisterEventSourceW (L"",L"Internet Explorer 8"): stub
003f:fixme:advapi:ReportEventA (0xcafe4242,0x0004,0x0000,0x400e1119,0x346320,0x0002,0x00000000,0x31bc34,(nil)): stub
003f:fixme:advapi:ReportEventW (0xcafe4242,0x0004,0x0000,0x400e1119,0x346320,0x0002,0x00000000,0xcc94b0,(nil)): stub
003f:fixme:advapi:DeregisterEventSource (0xcafe4242) stub
0038:fixme:ntdll:EtwUnregisterTraceGuids deadbeef: stub
Executing cd /home/VM-Konto/.local/share/wineprefixes/sm17/dosdevices/c:/windows/system32
0058:fixme:ntdll:EtwRegisterTraceGuidsA (0x6307379f, 0x630b1cf8, {0cfe0455-93ba-440d-a3fe-553973d0b723}, 1, 0x31e828, (null), (null), 0x630b1d00): stub
0058:fixme:ntdll:EtwRegisterTraceGuidsA (0x6307379f, 0x630b1d18, {797fabac-7b58-4796-b924-d51178a59ce4}, 1, 0x31e828, (null), (null), 0x630b1d20): stub
0058:fixme:ntdll:EtwEventRegister ({43d1a55c-76d6-4f7e-995c-64c711e5cafe}, 0x6309ce28, (nil), 0x630b1500) stub.
0058:fixme:ntdll:EtwRegisterTraceGuidsA (0x63073bcf, 0x630b3258, {9e3b3947-ca5d-4614-91a2-7b624e0e7244}, 1, 0x31ed6c, (null), (null), 0x630b3258): stub
0058:fixme:ntdll:EtwEventEnabled (deadbeef, 0x6302b3c0): stub
0058:fixme:ntdll:EtwEventEnabled (deadbeef, 0x63020688): stub
0058:fixme:ntdll:EtwEventUnregister (deadbeef) stub.
------------------------------------------------------
Wine-Fehler 25648 wird umgegangen -- Setting TabProcGrowth=0 to avoid hang
------------------------------------------------------
Executing wine regedit C:\windows\Temp\set-tabprocgrowth.reg
Executing cp -f /home/VM-Konto/.local/share/wineprefixes/sm17/dosdevices/c:/Program Files/Internet Explorer/ieproxy.dll /home/VM-Konto/.local/share/wineprefixes/sm17/dosdevices/c:/windows/system32
------------------------------------------------------
To start ie8, use the command "wine" 'C:\Program Files\Internet Explorer\iexplore'
------------------------------------------------------
Der Vorgang wurde erfolgreich abgeschlossen
Setting Windows version to default
Executing wine regedit C:\windows\Temp\set-winver.reg
------------------------------------------------------
Running /usr/bin/wineserver -w. This will hang until all wine processes in prefix=/home/VM-Konto/.local/share/wineprefixes/sm17 terminate
------------------------------------------------------
Executing w_do_call gdiplus_winxp
Executing load_gdiplus_winxp 
Executing mkdir -p /home/VM-Konto/.cache/winetricks/gdiplus_winxp
Executing cd /home/VM-Konto/.cache/winetricks/gdiplus_winxp
Downloading https://download.microsoft.com/download/1/4/6/1467c2ba-4d1f-43ad-8d9b-3e8bc1c6ac3d/NDP1.0sp2-KB830348-X86-Enu.exe to /home/VM-Konto/.cache/winetricks/gdiplus_winxp
--2020-04-07 14:09:06--  https://download.microsoft.com/download/1/4/6/1467c2ba-4d1f-43ad-8d9b-3e8bc1c6ac3d/NDP1.0sp2-KB830348-X86-Enu.exe
Auflösen des Hostnamens download.microsoft.com (download.microsoft.com)… 23.54.115.197, 2a02:26f0:3000:289::e59, 2a02:26f0:3000:287::e59, ...
Verbindungsaufbau zu download.microsoft.com (download.microsoft.com)|23.54.115.197|:443 … verbunden.
HTTP-Anforderung gesendet, auf Antwort wird gewartet … 200 OK
Länge: 1144608 (1.1M) [application/octet-stream]
Wird in »NDP1.0sp2-KB830348-X86-Enu.exe« gespeichert.

NDP1.0sp2-KB830348-X86-Enu.ex 100%[===============================================>]   1.09M  --.-KB/s    in 0.1s    

2020-04-07 14:09:07 (10.3 MB/s) - »NDP1.0sp2-KB830348-X86-Enu.exe« gespeichert [1144608/1144608]

Executing cd /home/VM-Konto
Executing cd /home/VM-Konto/.cache/winetricks/gdiplus_winxp
Executing cabextract -q -d /home/VM-Konto/.local/share/wineprefixes/sm17/dosdevices/c:/windows/temp -F FL_gdiplus_dll_____X86.3643236F_FC70_11D3_A536_0090278A1BB8 /home/VM-Konto/.cache/winetricks/gdiplus_winxp/NDP1.0sp2-KB830348-X86-Enu.exe
Executing cp /home/VM-Konto/.local/share/wineprefixes/sm17/dosdevices/c:/windows/temp/FL_gdiplus_dll_____X86.3643236F_FC70_11D3_A536_0090278A1BB8 /home/VM-Konto/.local/share/wineprefixes/sm17/dosdevices/c:/windows/system32/gdiplus.dll
Using native override for following DLLs: gdiplus
Executing wine regedit C:\windows\Temp\override-dll.reg
Executing w_do_call tahoma
Executing load_tahoma 
Executing mkdir -p /home/VM-Konto/.cache/winetricks/tahoma
Executing cd /home/VM-Konto/.cache/winetricks/tahoma
Downloading https://downloads.sourceforge.net/corefonts/OldFiles/IELPKTH.CAB to /home/VM-Konto/.cache/winetricks/tahoma
--2020-04-07 14:09:08--  https://downloads.sourceforge.net/corefonts/OldFiles/IELPKTH.CAB
Auflösen des Hostnamens downloads.sourceforge.net (downloads.sourceforge.net)… 216.105.38.13
Verbindungsaufbau zu downloads.sourceforge.net (downloads.sourceforge.net)|216.105.38.13|:443 … verbunden.
HTTP-Anforderung gesendet, auf Antwort wird gewartet … 302 Found
Platz: https://master.dl.sourceforge.net/project/corefonts/OldFiles/IELPKTH.CAB [folgend]
--2020-04-07 14:09:08--  https://master.dl.sourceforge.net/project/corefonts/OldFiles/IELPKTH.CAB
Auflösen des Hostnamens master.dl.sourceforge.net (master.dl.sourceforge.net)… 216.105.38.12
Verbindungsaufbau zu master.dl.sourceforge.net (master.dl.sourceforge.net)|216.105.38.12|:443 … verbunden.
HTTP-Anforderung gesendet, auf Antwort wird gewartet … 200 OK
Länge: 411248 (402K) [application/octet-stream]
Wird in »IELPKTH.CAB« gespeichert.

IELPKTH.CAB                   100%[===============================================>] 401.61K   475KB/s    in 0.8s    

2020-04-07 14:09:10 (475 KB/s) - »IELPKTH.CAB« gespeichert [411248/411248]

Executing cd /home/VM-Konto
Executing cabextract -q -d /home/VM-Konto/.local/share/wineprefixes/sm17/dosdevices/c:/windows/temp /home/VM-Konto/.cache/winetricks/tahoma/IELPKTH.CAB
/home/VM-Konto/.cache/winetricks/tahoma/IELPKTH.CAB: WARNING; possible 5592 extra bytes at end of file.
------------------------------------------------------
Running /usr/bin/wineserver -w. This will hang until all wine processes in prefix=/home/VM-Konto/.local/share/wineprefixes/sm17 terminate
------------------------------------------------------
Executing wine regedit C:\windows\Temp\_register-font.reg
000b:fixme:winediag:start_process Wine Staging 5.2 is a testing version containing experimental patches.
000b:fixme:winediag:start_process Please mention your exact version when filing bug reports on winehq.org.
000b:fixme:ntdll:EtwRegisterTraceGuidsA (0x6307379f, 0x630b1cf8, {0cfe0455-93ba-440d-a3fe-553973d0b723}, 1, 0x32e828, (null), (null), 0x630b1d00): stub
000b:fixme:ntdll:EtwRegisterTraceGuidsA (0x6307379f, 0x630b1d18, {797fabac-7b58-4796-b924-d51178a59ce4}, 1, 0x32e828, (null), (null), 0x630b1d20): stub
000b:fixme:ntdll:EtwEventRegister ({43d1a55c-76d6-4f7e-995c-64c711e5cafe}, 0x6309ce28, (nil), 0x630b1500) stub.
000b:fixme:ntdll:EtwEventRegister ({9e3b3947-ca5d-4614-91a2-7b624e0e7244}, 0x63073c41, 0x630b3258, 0x630b3180) stub.
000b:fixme:ntdll:EtwEventEnabled (deadbeef, 0x6302b3c0): stub
000b:fixme:ntdll:EtwEventEnabled (deadbeef, 0x63020688): stub
000b:fixme:ntdll:EtwEventUnregister (deadbeef) stub.
Executing cp /home/VM-Konto/.local/share/wineprefixes/sm17/dosdevices/c:/windows/temp/_register-font.reg /tmp/winetricks.jPm0QSzt/_reg_a6378261_7529.reg
Executing wine regedit C:\windows\Temp\_register-font.reg
Executing cp /home/VM-Konto/.local/share/wineprefixes/sm17/dosdevices/c:/windows/temp/_register-font.reg /tmp/winetricks.jPm0QSzt/_reg_2381d506_7529.reg
Executing wine regedit C:\windows\Temp\_register-font.reg
Executing cp /home/VM-Konto/.local/share/wineprefixes/sm17/dosdevices/c:/windows/temp/_register-font.reg /tmp/winetricks.jPm0QSzt/_reg_21e53125_7529.reg
Executing wine regedit C:\windows\Temp\_register-font.reg
Executing cp /home/VM-Konto/.local/share/wineprefixes/sm17/dosdevices/c:/windows/temp/_register-font.reg /tmp/winetricks.jPm0QSzt/_reg_4779f8a8_7529.reg
Executing w_do_call fontsmooth=rgb
Executing load_fontsmooth rgb
Setting font smoothing to rgb
Executing wine regedit C:\windows\Temp\fontsmooth.reg
Executing w_do_call winxp
Executing load_winxp 
Setting Windows version to winxp
Executing wine regedit C:\windows\Temp\set-winver.reg
------------------------------------------------------
Running /usr/bin/wineserver -w. This will hang until all wine processes in prefix=/home/VM-Konto/.local/share/wineprefixes/sm17 terminate
------------------------------------------------------
[9388:9388:0407/140919.715340:ERROR:edid_parser.cc(102)] Too short EDID data: manufacturer id
libva error: va_getDriverName() failed with unknown libva error,driver_name=(null)
[9416:9416:0407/140919.849457:ERROR:vaapi_wrapper.cc(481)] vaInitialize failed: unknown libva error
[9416:9416:0407/140920.103440:ERROR:viz_main_impl.cc(161)] Exiting GPU process due to errors during initialization
libva error: va_getDriverName() failed with unknown libva error,driver_name=(null)
[9436:9436:0407/140920.189297:ERROR:vaapi_wrapper.cc(481)] vaInitialize failed: unknown libva error
------------------------------------------------------
Bitte laden Sie sm17inst.exe von https://super-memo.com runter, stellen Sie's in /home/VM-Konto/.cache/winetricks/supermemo17, dann wiederholen Sie dieses Kommando.
------------------------------------------------------
VM-Konto@mx:~
$ [9388:9458:0407/140921.582666:ERROR:dat_file_util.cc(22)] GetDATFileData: the dat file is not found or corrupted /home/VM-Konto/.config/BraveSoftware/Brave-Browser/jmomcjcilfpbaaklkifaijjcnancamde/1.0.537/rs-E71426E7-E898-401C-A195-177945415F38.dat
[9388:9388:0407/140921.679707:ERROR:ad_block_base_service.cc(225)] Could not obtain ad block data
[9388:9455:0407/140921.848144:ERROR:rewards_service_impl.cc(206)] Failed to read file: /home/VM-Konto/.config/BraveSoftware/Brave-Browser/Default/ledger_state
[9439:9439:0407/140921.899301:ERROR:ledger_impl.cc(107)] Failed to initialize wallet
Failed to parse JSON adblock resources: EOF while parsing a value at line 1 column 0
[9388:9388:0407/140922.691078:ERROR:CONSOLE(1)] "[Shields]: Can't request shields panel data. Error: No tab url specified", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[9388:9388:0407/140925.131551:ERROR:CONSOLE(1)] "[Shields]: Can't request shields panel data. Error: No tab url specified", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[9556:58:0407/140932.051888:ERROR:child_process_sandbox_support_impl_linux.cc(79)] FontService unique font name matching request did not receive a response.
[9556:58:0407/140932.052379:ERROR:child_process_sandbox_support_impl_linux.cc(79)] FontService unique font name matching request did not receive a response.
[9556:58:0407/140932.052992:ERROR:child_process_sandbox_support_impl_linux.cc(79)] FontService unique font name matching request did not receive a response.
[9556:58:0407/140932.053364:ERROR:child_process_sandbox_support_impl_linux.cc(79)] FontService unique font name matching request did not receive a response.
$ [9388:9388:0407/140956.601685:ERROR:CONSOLE(0)] "Unchecked runtime.lastError: Cannot access contents of url "chrome://history/". Extension manifest must request permission to access this host.", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/_generated_background_page.html (0)
[9388:9388:0407/140956.706043:ERROR:CONSOLE(4)] "Polymer is already defined.", source: chrome://history/app.crisper.js (4)
[9679:147:0407/140956.824976:ERROR:child_process_sandbox_support_impl_linux.cc(79)] FontService unique font name matching request did not receive a response.
[9679:147:0407/140956.825422:ERROR:child_process_sandbox_support_impl_linux.cc(79)] FontService unique font name matching request did not receive a response.
[9705:158:0407/141000.488570:ERROR:child_process_sandbox_support_impl_linux.cc(79)] FontService unique font name matching request did not receive a response.
[9705:158:0407/141000.489145:ERROR:child_process_sandbox_support_impl_linux.cc(79)] FontService unique font name matching request did not receive a response.
[9705:158:0407/141000.489780:ERROR:child_process_sandbox_support_impl_linux.cc(79)] FontService unique font name matching request did not receive a response.
[9705:158:0407/141000.490176:ERROR:child_process_sandbox_support_impl_linux.cc(79)] FontService unique font name matching request did not receive a response.
[9705:158:0407/141000.490887:ERROR:child_process_sandbox_support_impl_linux.cc(79)] FontService unique font name matching request did not receive a response.
[9705:158:0407/141000.491280:ERROR:child_process_sandbox_support_impl_linux.cc(79)] FontService unique font name matching request did not receive a response.
[9705:158:0407/141000.491906:ERROR:child_process_sandbox_support_impl_linux.cc(79)] FontService unique font name matching request did not receive a response.
[9705:158:0407/141000.492299:ERROR:child_process_sandbox_support_impl_linux.cc(79)] FontService unique font name matching request did not receive a response.
[9388:9388:0407/141020.566581:ERROR:gcm_channel_status_request.cc(145)] GCM channel request failed.
```

**Comment by alessivs** [comment]
All is good. The installer doesn't really wait for you to copy the file. If there is a delay, it's because it's shutting down processes (wineserver, mainly).

As asked, you have to manually copy the file there, but only once, and this only applies to the commercial editions of SuperMemo (currently, 17 and 18); supermemo-wine will automatically download SuperMemo 16 and below. This may change as SuperMemo World officially makes the installers publicly available (currently, they don't for the commercial editions).

After you copy the file to the specified location, all you have to do is re-run the command you entered first (*«dann wiederholen Sie dieses Kommando»*). By repeating the command, it won't install what's already installed, and will effectively resume the installation by installing SuperMemo.

So repeat the command (if it's in your shell history you could maybe just press the up arrow). Good luck!

**Comment by Golddouble** [comment]
Thank you.
I repeated the command. This is what I got:

Extract from my terminal log:
```
Running /usr/bin/wineserver -w. This will hang until all wine processes in prefix=/home/VM-Konto/.local/share/wineprefixes/sm17 terminate
------------------------------------------------------
------------------------------------------------------
sha256sum mismatch! Rename /home/VM-Konto/.cache/winetricks/supermemo17/sm17inst.exe and try again.
------------------------------------------------------
```
Not sure what I have to rename.
This is my original sm17inst.exe, that is 524.4 KB and must load the rest from the internet.
 (?)

**Comment by alessivs** [comment]
What does the following return?

```
sha256sum /home/VM-Konto/.cache/winetricks/supermemo17/sm17inst.exe
```

**Comment by Golddouble** [comment]
```
VM-Konto@mx:~
$ sha256sum /home/VM-Konto/.cache/winetricks/supermemo17/sm17inst.exe
2add9eebc8398847e9a82b711ff88cd04fcba877700dc0f086630701bd98b5c4  /home/VM-Konto/.cache/winetricks/supermemo17/sm17inst.exe
VM-Konto@mx:~
```

**Comment by alessivs** [comment]
The checksum differs from my copy of `sm17inst.exe`. If you are sure of the file's integrity, use [this verb](https://gist.github.com/alessivs/54a3922a2cb9cae0e998a779659616f4) instead. I'll try to re-download my own order of sm17 to see if anything changed. Thanks for sticking with the install process thus far.

**Comment by Golddouble** [comment]
Wow, very impressive. I got it. Thank you.

**Question 1:**
I have installed it in a virtual machine first (for testing). Later I will install it in my working system. Can I use the version 2 of your "spermemo17.verb" (https://gist.github.com/alessivs/54a3922a2cb9cae0e998a779659616f4) from the beginning of the installation? 

**Question 2:**
There seems to be something wrong with the toolbar in wine-version. Some icons are missing, other are separated through space.

Windows:
https://www.bilder-upload.eu/bild-b9f662-1586348389.png.html

Wine:
https://www.bilder-upload.eu/bild-9c64d1-1586348373.png.html

**Comment by alessivs** [comment]
Q1: Yes.

Q2: I know! Nothing we can do, but I have made a patch to remedy that. See: https://github.com/alessivs/supermemo-miscellany#compact-navigation-bar

**Comment by alessivs** [comment]
Whoops, I don't have the patch available for 17.40. I'll make one and publish tomorrow-ish.

**Comment by alessivs** [comment]
Closing this issue for now. If you wish you can join this gitter room for general questions on SM on Wine: https://gitter.im/supermemobits/supermemo-wine , or open a new issue.

**Comment by Golddouble** [comment]
OK. Thank you very much for your kind support and your patient.
Golddouble

**Comment by alessivs** [comment]
- README to be improved to incorporate instructions for commercial SuperMemo installers (#6)
- Installer checksums to be revised (#8)
- Patches for SuperMemo 17.40 (alessivs/supermemo-miscellany#1)

**Comment by alessivs** [comment]
@Golddouble Check out https://github.com/alessivs/supermemo-miscellany#patches for a v17.40 patch to achieve a compact navigation bar.
